backend/corpora/lambdas/api/v1/collection.py

```diff
@@ -68,3 +68,17 @@ def create_collection(body: object, user: str):
 
 def get_collection_dataset(dataset_uuid: str):
     raise NotImplementedError
+
+
+def delete_collection(collection_uuid: str, visibility: str, user: str):
+    if visibility == CollectionVisibility.PUBLIC.name:
+        return make_response(jsonify("Can not delete a public dataset"), 405)
+    db_session = g.db_session
+    collection = Collection.get_collection(db_session, collection_uuid, visibility, include_tombstones=True)
+    if not collection:
+        raise ForbiddenHTTPException()
```

Comment:
Is Forbidden the most accurate exception we can throw here, it's a bit misleading since the message sent with it is "User is not authorized to access this resource."  Perhaps a 404 would be better?

Reply:
This is common practice to throw forbidden rather than 404. It prevents users from probing for used UUIDs.

Reply:
I think its actually meant to be a little misleading, my understanding is we send the same error message for a variety of errors so a malicious user cant tell from the error message which resources exist and they arent allowed to see and which just dont exist, but @Bento007 may have a better explanation/another opinion

Reply:
No that was a good explanation.